### ISSUE TITLE
Run tasks command iterates on all files.

### ISSUE THREAD

**Issue by tifabien** [phpro/grumphp#139]
**My configuration**
```yaml
# grumphp.yml
parameters:
    tasks:
        phpcsfixer:
            config_file: '.php_cs'
```

**.php_cs config file**

```php
<?php

return Symfony\CS\Config\Config::create()
    ->setUsingCache(true)
    ->level(Symfony\CS\FixerInterface::SYMFONY_LEVEL)
    ->fixers(array('ordered_use'))
    ->finder(
        Symfony\CS\Finder\DefaultFinder::create()
            ->in(__DIR__)
            ->notPath('web/app.php')
```

**Steps to reproduce:**
```sh
# execute run command
vendor/bin/grumphp run
```

**Result:**
The run command iterates on all project files, ignore the config file (it will be fixed in PHPcsfixer 2) and fails on web/app.php file

I Think that for the run command if we get a config file configured, we should not iterates on all files but just execute the ```sh './vendor/bin/php-cs-fixer' '--format=json' '--dry-run' '--config-file=.php_cs' '--verbose' 'fix'  ``` command

What do you think?

**Comment by veewee** [comment]
Hello @tifabien,

This is more a php-cs-fixer issue then a grumphp issue. We discussed it already in a previous issue: https://github.com/phpro/grumphp/issues/95

Currently we are only validating the changed files during a commit. If we change it to the full configuration a lot of additional files will be validated again. This wil not only slow the application down but also might throw exceptions on code that you are not commiting.

This is why we decided to mark the issue as a won't fix.  One thing that might be possible is to add a new task property `ignore_patterns` and change the task to ignore patterns. The downside of this approach is that you'll have to configure it twice.

When version 2 gets released we will propably have to write a new grumphp task for php-cs-fixer.

**Comment by tifabien** [comment]
Hi @veewee 

I agree for the git hooks command which must only check modified files and not all files in the project.

My question regards the grumphp run command used for CI servers commonly. Think we should not execute one php-cs-fixer command for each file in the project but only one single command for checking codestyle of the whole project.
This could improve performance of the task execution.

We could effectively improve php-cs-fixer task when v2 will be release but this could be also integrated with the V1.

**Comment by veewee** [comment]
Sounds good. Only when the configfile is set And the run command is triggered? Or Always during the run command?

**Comment by tifabien** [comment]
I Think we could do this always the run command is executed.

**Comment by veewee** [comment]
@tifabien, What if a user did not configured a config file?
You will get something like:

>   [LogicException]
>  You must call one of in() or append() methods before iterating over a Finder.

When nog config file is added, should the current directory by used?

**Comment by veewee** [comment]
Fixed in #142

**Comment by tifabien** [comment]
@veewee Thanks for the work! i'll test it asap and will let you know if I found issues!
